src/engine/filters.ts

```diff
@@ -0,0 +1,12 @@
+import { vectorDistance } from '../utils';
+
+export const includes = <T>(queryFeatures: T[], queryValue: T[]): boolean =>
+  queryFeatures.some((feature) => queryValue.indexOf(feature) !== -1);
+
+export const dotProduct = (
```

Comment:
dotProduct can be called vectorDistance here as it incorporates the threshold metric.